### PR TITLE
Persist sqlmode in load metadata and add sqlmode to forward master request

### PR DESCRIPTION
**Pull request by xy720**
ISSUE #2213 
This commit includes the following:
1. Persist sqlmode in load metadata so that it can be get to re-parse/re-analyze the querystmt when the routine/broker/stream load is replayed.
2. Add sqlmode as a param into forward master request so that non-leader node can transfer its sql mode to leader node when the request is send.
3. Modify the SqlScanner constructor, let sqlmode can be passed as a param to the scanner

### CODE REVIEW

File: fe/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java [hunk 1]
```diff
@@ -100,6 +105,13 @@ private BrokerLoadJob(long dbId, String label, BrokerDesc brokerDesc, String ori
         this.originStmt = Strings.nullToEmpty(originStmt);
         this.jobType = EtlJobType.BROKER;
         this.authorizationInfo = gatherAuthInfo();
+
+        if (ConnectContext.get() != null) {
+            SessionVariable var = ConnectContext.get().getSessionVariable();
+            sessionVariables.put(SessionVariable.SQL_MODE, Long.toString(var.getSqlMode()));
+        } else {
+            sessionVariables.put(SessionVariable.SQL_MODE, "0");
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java]: Use a defined variable to replace "0" here, such as SQLMode.DEFAULT

File: fe/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java [hunk 2]
```diff
@@ -518,6 +537,18 @@ public void readFields(DataInput in) throws IOException {
...
+        if (Catalog.getCurrentCatalogJournalVersion() >= FeMetaVersion.VERSION_66) {
+            int size = in.readInt();
+            for (int i = 0; i < size; i++) {
+                String key = Text.readString(in);
+                String value = Text.readString(in);
+                sessionVariables.put(key, value);
+            }
+        } else {
+            // old version of load does not have sqlmode, set it to default
+            sessionVariables.put(SessionVariable.SQL_MODE, "0");
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/load/loadv2/BrokerLoadJob.java]: Use a defined variable to replace "0" here

File: fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java [hunk 3]
```diff
@@ -389,6 +402,10 @@ public String getTimezone() {
         return value;
     }
 
+    public Map<String, String> getSessionVariables() {
```
Comment (morningman) [inline fe/src/main/java/org/apache/doris/load/routineload/RoutineLoadJob.java]: If there is no other place to call this get method, remove it.

File: gensrc/thrift/FrontendService.thrift [hunk 4]
```diff
@@ -406,13 +406,14 @@ struct TMasterOpRequest {
...
     2: required string db
     3: required string sql 
-    4: optional Types.TResourceInfo resourceInfo
-    5: optional string cluster
-    6: optional i64 execMemLimit
-    7: optional i32 queryTimeout
-    8: optional string user_ip
-    9: optional string time_zone
-    10: optional i64 stmt_id
+    4: required i64 sqlMode
```
Comment (morningman) [inline gensrc/thrift/FrontendService.thrift]: Add this sqlMode to the end. Better to keep the origin variables order.

**Review by morningman — CHANGES_REQUESTED** [review]

**Review by morningman — APPROVED** [review]
LGTM

File: gensrc/thrift/FrontendService.thrift [hunk 5]
```diff
@@ -413,6 +413,7 @@ struct TMasterOpRequest {
     8: optional string user_ip
     9: optional string time_zone
     10: optional i64 stmt_id
+    11: required i64 sqlMode
```
Comment (imay) [inline gensrc/thrift/FrontendService.thrift]: New added field should be optional.

**Pull request merged**